### PR TITLE
2.x.x mongokitten tutorial

### PR DESCRIPTION
**Pull request by Joannis**
- Needs to use swift-server/hummingbird-openapi once merged
- Needs to use the alpha once tagged

### CODE REVIEW

File: todos-mongokitten-openapi/Package.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import PackageDescription
+
+let package = Package(
+    name: "todos-mongokitten-openapi",
+    platforms: [ .macOS(.v14) ],
+    dependencies: [
+        .package(url: "https://github.com/hummingbird-project/hummingbird.git", branch: "2.x.x"),
+        .package(url: "https://github.com/apple/swift-argument-parser.git", from: "1.0.0"),
+        .package(url: "https://github.com/orlandos-nl/mongokitten.git", from: "7.0.0"),
+        .package(url: "https://github.com/Joannis/swift-openapi-hummingbird.git", branch: "feature/jo/support-hb-2.x.x"),
```
Comment (adam-fowler) [inline todos-mongokitten-openapi/Package.swift:13]:
```suggestion
        .package(url: "https://github.com/Joannis/swift-openapi-hummingbird.git", branch: "2.x.x"),
```

File: todos-mongokitten-openapi/Sources/todos-mongokitten-openapi/API.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
+@preconcurrency import MongoKitten
+import OpenAPIRuntime
+import TodosOpenAPI
+
+struct TodoModel: Codable {
+    let _id: ObjectId
+    let items: [String]
```
Comment (adam-fowler) [inline todos-mongokitten-openapi/Sources/todos-mongokitten-openapi/API.swift:7]: What is `items`? The todo doesn't seem to have a title, or status. Is this somehow store in `items`

File: todos-mongokitten-openapi/Sources/TodosOpenAPI/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
+openapi: '3.1.0'
```
Comment (adam-fowler) [inline todos-mongokitten-openapi/Sources/TodosOpenAPI/openapi.yaml:1]: Is it considered good practice to separate the OpenAPI generated code into another module?

**Review by adam-fowler — CHANGES_REQUESTED** [review]
This looks really good. Nice alternative to the other todo samples, as you have used openapi.
- As an example it needs more comments. 
- There needs to be README.md explaining what the sample is and how to run it. I guess you need a locally running mongodb instance. Does it need some setup?
- Also the README in the root needs updated, to include a link

File: todos-mongokitten-openapi/Sources/TodosOpenAPI/openapi.yaml (hunk 3 above)
Comment (Joannis) [reply todos-mongokitten-openapi/Sources/TodosOpenAPI/openapi.yaml:1]: I've found it to be good practice, as Swift tends to generate errors for the types in that file if it's not in a dependency. It gets confused about the file not existing

File: todos-mongokitten-openapi/Package.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                .product(name: "OpenAPIRuntime", package: "swift-openapi-runtime"),
+            ],
+            plugins: [.plugin(name: "OpenAPIGenerator", package: "swift-openapi-generator")]
+        ),
+        .executableTarget(
+            name: "todos-mongokitten-openapi",
+            dependencies: [
+                .product(name: "ArgumentParser", package: "swift-argument-parser"),
+                .product(name: "Hummingbird", package: "hummingbird"),
+                .product(name: "HummingbirdFoundation", package: "hummingbird"),
```
Comment (adam-fowler) [inline todos-mongokitten-openapi/Package.swift:30]:
```suggestion
```

**Review by adam-fowler — CHANGES_REQUESTED** [review]
Just needs a readme and the HummingbirdFoundation dependency removed and its good

**Review by adam-fowler — APPROVED** [review]

**Pull request merged**